nmosquery/couchbasewatcher.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019 British Broadcasting Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gevent
+from nmoscommon.timestamp import Timestamp
+
+DEFAULT_START_INTERVAL = 15  # TODO: config var? maybe accessed via registry object
+POLL_RATE = 5
+
+
+class CouchbaseWatcher(gevent.Greenlet):
+    def __init__(self, registry, handler, logger):
+        gevent.Greenlet.__init__(self)
+        self.registry = registry
+        self.handler = handler
+        self.logger = logger
+        self.events = None  # TBD if some sort of array of events makes any sense
+        self.running = False
+
+    def _run(self):
+        self.logger.writeDebug('couchbasewatcher: running')
+        self.running = True
+
+        current_time = Timestamp.get_time().to_nanosec()
+        start_time = current_time - Timestamp(sec=(DEFAULT_START_INTERVAL * 60)).to_nanosec()
```

Comment:
`current_time` and `start_time` should be updated below `while self.running` (line 37) so that the time in the query isn't stale.